xmtp_mls/src/groups/mod.rs

```diff
@@ -397,10 +397,12 @@ where
         permissions_policy_set: PolicySet,
         opts: GroupMetadataOptions,
     ) -> Result<Self, GroupError> {
+        assert!(conversation_type != ConversationType::Dm);
```

Comment:
Sort of unrelated change to protect future implementers - we have a separate method for creating DM's, so it would be a mistake to call this method with a `conversation_type` of Dm. Already audited all current usages of this method.